cmd/version.go

```diff
@@ -24,16 +25,70 @@ func versionString() string {
 	return v
 }
 
-func getCmdVersion(_ *state.GlobalState) *cobra.Command {
-	// versionCmd represents the version command.
-	return &cobra.Command{
+type versionCmd struct {
+	gs     *state.GlobalState
+	isJSON bool
+}
+
+func (c *versionCmd) run(cmd *cobra.Command, _ []string) error {
+	if !c.isJSON {
+		root := cmd.Root()
+		root.SetArgs([]string{"--version"})
+		_ = root.Execute()
+		return nil
+	}
+
+	details := consts.VersionDetails()
+	if exts := ext.GetAll(); len(exts) > 0 {
+		type extInfo struct {
+			Module  string   `json:"module"`
+			Version string   `json:"version"`
+			Imports []string `json:"imports"`
+		}
+
+		ext := make(map[string]extInfo)
+		for _, e := range exts {
+			if v, ok := ext[e.Path]; ok {
+				v.Imports = append(v.Imports, e.Name)
+				ext[e.Path] = v
+				continue
+			}
+
+			ext[e.Path] = extInfo{
+				Module:  e.Path,
+				Version: e.Version,
+				Imports: []string{e.Name},
+			}
+		}
+
+		list := make([]extInfo, 0, len(ext))
+		for _, v := range ext {
+			list = append(list, v)
+		}
+
+		details["extensions"] = list
+	}
+
+	jsonDetails, err := json.Marshal(details)
+	if err != nil {
+		return fmt.Errorf("failed produce a JSON version details: %w", err)
+	}
+
+	_, err = fmt.Fprintln(c.gs.Stdout, string(jsonDetails))
+	return err
+}
+
+func getCmdVersion(gs *state.GlobalState) *cobra.Command {
+	versionCmd := &versionCmd{gs: gs}
+
+	cmd := &cobra.Command{
 		Use:   "version",
 		Short: "Show application version",
 		Long:  `Show the application version and exit.`,
-		Run: func(cmd *cobra.Command, _ []string) {
-			root := cmd.Root()
-			root.SetArgs([]string{"--version"})
-			_ = root.Execute()
-		},
+		RunE:  versionCmd.run,
```

Comment:
I guess this means that now `k6 --version` and `k6 version` will output different things.

Reply:
> I guess this means that now k6 --version and k6 version will output different things.

That [behavior remains](https://github.com/grafana/k6/pull/4093/files#diff-7cb26babce207ed9c17e9dd3ead0b66d204da3cf4d11649412c197defbb3029cR34-R39) for a non JSON output